framework/src/bundle/CoreBundleContext.h

```diff
@@ -56,10 +56,22 @@ as specified in the OSGi R4.2 specifications.
 
 #include <map>
 #include <ostream>
+#include <shared_mutex>
 #include <string>
 
 namespace cppmicroservices
 {
+    using shared_lock = std::shared_mutex;
+    using WriteLock = std::unique_lock<shared_lock>;
+    using ReadLock = std::shared_lock<shared_lock>;
```

Comment:
the use of the word `lock` for a mutex and for a C++ `*_lock` type is confusing. I recommend removing `using shared_lock = std::shared_mutex;` and reference `std::shared_mutex`. The `using` statement is unnecessary.

Reply:
I agree. Fixed